docs/guides/publish_fetch_packages.md

```diff
@@ -0,0 +1,177 @@
+In this guide, you will learn to:
+
+  * Manage the life cycle of packages with the framework.
+  * Publish and retrieve packages from local and remote registries.
+
+## How the local and remote registries work
+There are three main types of packages:
+
+  * services,
+  * agents, and
+  * components (which include connections, contracts, protocols and skills).
+
+Services use agent packages, and agents use component packages. Packages are developed locally and stored in a registry to be retrieved and reused in a later stage. Every package has a **public package ID** that follows the naming convention `<author_name>/<package_name>:<version>`.
+
+
+The {{open_autonomy}} framework supports three types of registries:
+
+  * local,
+  * remote [IPFS](https://ipfs.io/), and
+  * remote HTTP.
+
+You can browse the [list of default packages](../package_list.md) of the {{open_autonomy}} framework available on the default remote IPFS registry.
+
+Packages live in a different space when being used by the developer, and they are published in a registry using different commands. The table below presents a summary.
+
+| Package type | Location while being used           | Command to publish in a registry | Command to retrieve from a registry |
+|--------------|-------------------------------------|----------------------------------|-------------------------------------|
+| Service      | Independent folder                  | `autonomy publish`               | `autonomy fetch`                    |
+| Agent        | Independent folder                  | `autonomy publish`               | `autonomy fetch`                    |
+| Component    | `/vendor` folder inside agent folder | `autonomy push`                  | `autonomy add`                      |
+
+
+
+The figure below shows a typical setup, where you might identify some common traits with the Git methodology. Within the workspace folder, the arrows in the diagram point to the locations where the different operations can be executed in the local machine.
+
+<figure markdown>
+![](../images/package_management.svg){ width="85%" height="85%" style="display: block; margin: 0 auto" }
+<figcaption>Overview of the package management flow with the Open Autonomy framework</figcaption>
+</figure>
+
+!!! warning "Important"
+    Do not confuse the **local registry** with a local **workspace folder**. The local registry is simply a repository to store finalized packages, mimicking the role of a remote registry on the local machine.
+
+    For example, you can develop a component package directly in the local registry, or you can develop it in an agent within a workspace folder, and push the component later in the local registry. Then, you can retrieve that component from the registry for use in another agent.
+
+    **Regardless of the methodology that you choose to develop your packages, it is important to have a clear separation of concerns in mind so that they can be reused in other agent projects.**
```

Comment:
The package consistency checks, run when loading the package are helping with this.